circuits/fp.circom

```diff
@@ -207,6 +207,12 @@ template SignedCheckCarryModToZero(n, k, overflow, p){
         X_range_checks[i] = Num2Bits(n+1);
         X_range_checks[i].in <== X[i] + (1<<n); // X[i] should be between [-2^n, 2^n)
     }
+
+    signal remainder[50];
+    for (var i=0; i<k; i++) {
+        remainder[i] <-- Xvar[1][i];
+        remainder[i] === 0;
```

Comment:
I think the constraint is not necessary because later `mod_check.Y[i] <== 0` is essentially imposing the same constraint. In other words, `mod_check` is constrain that `Xvar[0] * p + 0 === in`, so we never actually need to know what `Xvar[1]` is.

Reply:
p = 4002409555221667393417789825735904156556882819939007885332058136124031650490837864442687629129015664037894272559787

fp in bn254 is 21888242871839275222246405745257275088696311157297823662689037894645226208583

if in = 21888242871839275222246405745257275088696311157297823662689037894645226208583

X = in / p = 0
remainder = in%p = 21888242871839275222246405745257275088696311157297823662689037894645226208583

X*p+0 = 0 = in mod fpBn254

Reply:
What limb size are you using? `CheckCarryToZero` should be able to see that `in != 0` if there's more than 1 limb